### PR TITLE
chore(deps): migrate to jsonrpsee 0.22

### PR DESCRIPTION
**Pull request by mattsse**
the latest release of jsonrpsee removes a bunch of logging functionality which breaks our current rpc metrics setup...

not sure how to proceed with this yet

### CODE REVIEW

**Comment by Rjected** [comment]
The changelog for 21.0 mentions that a middleware could be used, maybe we could use that
https://github.com/paritytech/jsonrpsee/releases/tag/v0.21.0

```rust
let rpc_middleware = RpcServiceBuilder::new().rpc_logger(1024);
let server = Server::builder().set_rpc_middleware(rpc_middleware).build("127.0.0.1:0").await?;
```

**Comment by mattsse** [comment]
Okay finally pulled through.

There's currently one limitations wrt rpc metrics if both HTTP and ws are enabled on the same port. this will hopefully unblocked via a jsonrpsee feature

File: crates/rpc/ipc/src/server/mod.rs [hunk 1]
```diff
@@ -483,14 +483,16 @@ impl<B, L> Builder<B, L> {
...
+    /// Add a logger to the builder.
+    pub fn set_logger(self) -> Builder<B> {
+        todo!()
+
+        // Builder {
+        //     settings: self.settings,
+        //     logger,
+        //     id_provider: self.id_provider,
+        //     service_builder: self.service_builder,
+        // }
```
Comment (onbjerg) [inline crates/rpc/ipc/src/server/mod.rs:495]: follow up?

File: crates/rpc/rpc-builder/src/lib.rs [hunk 2]
```diff
@@ -1736,11 +1737,13 @@ impl RpcServerConfig {
...
 
         if let Some(builder) = self.ipc_server_config {
-            let metrics = modules.ipc.as_ref().map(RpcServerMetrics::new).unwrap_or_default();
+            // let metrics = modules.ipc.as_ref().map(RpcRequestMetrics::new).unwrap_or_default();
             let ipc_path = self
                 .ipc_endpoint
                 .unwrap_or_else(|| Endpoint::new(DEFAULT_IPC_ENDPOINT.to_string()));
-            let ipc = builder.set_logger(metrics).build(ipc_path.path())?;
+            let ipc = builder
+                // .set_middleware(metrics)
```
Comment (onbjerg) [inline crates/rpc/rpc-builder/src/lib.rs:1745]: follow up?

File: crates/rpc/rpc-builder/src/metrics.rs [hunk 3]
```diff
@@ -91,60 +227,60 @@ struct RpcServerCallMetrics {
...
-        // capture request latency for this request/response pair
-        metrics.request_time_seconds.record(started_at.elapsed().as_secs_f64());
-        metrics.requests_finished.increment(1);
-    }
-
-    fn on_disconnect(&self, _remote_addr: SocketAddr, transport: TransportProtocol) {
-        self.inner.connection_metrics.get_metrics(transport).connections_closed.increment(1)
-    }
-}
+// impl Logger for RpcServerMetrics {
```
Comment (onbjerg) [inline crates/rpc/rpc-builder/src/metrics.rs:230]: same here

File: crates/rpc/rpc-engine-api/src/error.rs [hunk 4]
```diff
@@ -210,11 +210,11 @@ impl From<EngineApiError> for jsonrpsee_types::error::ErrorObject<'static> {
     }
 }
 
-impl From<EngineApiError> for jsonrpsee_core::error::Error {
-    fn from(error: EngineApiError) -> Self {
-        jsonrpsee_core::error::Error::Call(error.into())
-    }
-}
+// impl From<EngineApiError> for jsonrpsee_core::error::Error {
+//     fn from(error: EngineApiError) -> Self {
```
Comment (onbjerg) [inline crates/rpc/rpc-engine-api/src/error.rs:214]: rm?

File: crates/rpc/rpc-builder/src/lib.rs (hunk 2 above)
Comment (mattsse) [reply crates/rpc/rpc-builder/src/lib.rs:1745]: yep

**Review by onbjerg — APPROVED** [review]
looks good from my side

File: Cargo.lock [hunk 5]
```diff
@@ -5737,23 +5766,23 @@ dependencies = [
  "once_cell",
  "percent-encoding",
  "pin-project-lite",
- "rustls",
- "rustls-pemfile",
+ "rustls 0.21.10",
+ "rustls-pemfile 1.0.4",
```
Comment (DaniPopes) [inline Cargo.lock:5770]: deduped with https://github.com/paradigmxyz/reth/pull/7018 on a future jsonrpsee version

**Review by DaniPopes — APPROVED** [review]
lgmt

**Pull request merged**